### PR TITLE
Adding GENIE Tree index to SRTrueInteraction [release/SBN2024A]

### PR DESCRIPTION
**Pull request by jedori0228**
This is a copy of https://github.com/SBNSoftware/sbnanaobj/pull/128  (on develop) or https://github.com/SBNSoftware/sbnanaobj/pull/124  (on release/SBN2023A_NuMI).

Adding a new `size_t genie_evtrec_idx` variable into `SRTrueInteraction` which corresponds to the index of the corresponding `genie::EventRecord`.
Also added `std::uint32_t sourceNameHash`, where we save the truncated hash of `sourceName`; this can be used when matching to auxiliary trees other than `recTree`.

### CODE REVIEW

**Review by jacoblarkin — APPROVED** [review]

File: sbnanaobj/StandardRecord/SRHeader.h [hunk 1]
```diff
@@ -56,6 +57,7 @@ namespace caf
 
       std::string    sourceName; ///< Name of the file or source this event comes from.
       unsigned int   sourceIndex = NoSourceIndex; ///< Index of this event within the source (zero-based).
+      std::uint32_t  sourceNameHash; ///< hash of sourceName, std::hash<std::string>(sourceName), then truncated to std::uint32_t. Should be 32-bit integer to be used as TTreeIndex (https://root.cern/doc/master/classTTreeIndex.html#a08aac749ab22fd5c8ab792a0061a4b0f)
```
Comment (absolution1) [inline sbnanaobj/StandardRecord/SRHeader.h:60]: Could this variable be initialised with something (like 99999) in the constructor list?

**Review by absolution1 — CHANGES_REQUESTED** [review]
Sorry this is a bit of a drive-by review.

The added index should be initialised with a sensible default in the constructor list.  We saw an issue with this var taking on random values, causing errors in the CI.

Could a default be added in?

Thank you!

File: sbnanaobj/StandardRecord/SRHeader.h (hunk 1 above)
Comment (jedori0228) [reply sbnanaobj/StandardRecord/SRHeader.h:60]: Thanks for the comments! I set the default as 0 in the new commit.

**Review by absolution1 — APPROVED** [review]
Thanks @jedori0228 .   The other PR addresses the raised issue.

**Pull request merged**